lib/core/src/Cardano/Wallet/Api/Server.hs

```diff
@@ -265,10 +266,14 @@ listAddresses
     -> ApiT WalletId
     -> Maybe (ApiT AddressState)
     -> Handler [ApiAddress t]
-listAddresses w (ApiT wid) _ = do
+listAddresses w (ApiT wid) stateFilter = do
     addrs <- liftHandler $ W.listAddresses w wid
-    return $ coerceAddress <$> addrs
+    return $ coerceAddress <$> filter filterCondition addrs
```

Comment:
Not strictly relevant to this PR, but maybe in future we should cache addresses and their status in the database?

Reply:
@rvl How do you envision caching here exactly :thinking:  ?